### PR TITLE
Add decision page for scheduling billing jobs

### PR DESCRIPTION
**Pull request by HappyTetrahedron**
## Checklist

- [x] Try to isolate changes into separate PRs (to build a better changelog).
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `change`, `decision`, `requirement/quality`, `requirement/functional`, `dependency`
      as they show up in the changelog

### CODE REVIEW

File: docs/modules/ROOT/pages/explanation/decisions/billing-scheduling-monitoring.adoc [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+* Optimizing the ETL process itself
+* Detecting missing usage data directly in the target ERP system
+
+== Proposals
+
+=== Option 1: Cronjobs per Product
+
+This describes the current implementation, as a baseline for comparison:
+
+A Project Syn Component produces a kube cronjob for each product that requires processing.
```
Comment (simu) [inline docs/modules/ROOT/pages/explanation/decisions/billing-scheduling-monitoring.adoc:28]:
```suggestion
A Project Syn component produces a Kubernetes cronjob for each product that requires processing.
```

File: docs/modules/ROOT/pages/explanation/decisions/billing-scheduling-monitoring.adoc [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+.Con
+
+* Re-running a failed job does not produce expected results - it will process billing data for the current time, and not the time at which the job was supposed to run.
+* Even though the system can recover from transient errors, they still generate an alert in any case, and an investigation is required to determine whether manual intervention is necessary.
+* It is not easily possible to run the ETL process for an arbitrary timeframe.
+* Running on the order of 100 jobs each hour is taxing our infrastructure.
+
+=== Option 2: Spawner Cronjob
+
+A Project Syn Component produces a single cronjob and provides it with a list of products for which the ETL process needs to run (including associated parameters).
```
Comment (simu) [inline docs/modules/ROOT/pages/explanation/decisions/billing-scheduling-monitoring.adoc:52]:
```suggestion
A Project Syn component produces a single cronjob and provides it with a list of products for which the ETL process needs to run (including associated parameters).
```

File: docs/modules/ROOT/pages/explanation/decisions/billing-scheduling-monitoring.adoc [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+The cronjob runs hourly.
+On each run, the cronjob generates a new job for each product which in turn runs the ETL process.
```
Comment (simu) [inline docs/modules/ROOT/pages/explanation/decisions/billing-scheduling-monitoring.adoc:54]: Would it be feasible to generate the jobs in such a fashion that we can circumvent [thundering herd](https://en.wikipedia.org/wiki/Thundering_herd_problem) issues? For example by injecting configurable randomized delays (on the order of a few minutes) to smear the actual work of the jobs over 15 minutes or so.

This doesn't necessarily need to be outlined here, but we may want to consider this for the implementation.

**Review by simu — APPROVED** [review]
LGTM overall. Inline comments for some formatting/spelling nitpicks, and a random implementation idea that's probably too detailed for the decision page.

File: docs/modules/ROOT/pages/explanation/decisions/billing-scheduling-monitoring.adoc (hunk 3 above)
Comment (HappyTetrahedron) [reply docs/modules/ROOT/pages/explanation/decisions/billing-scheduling-monitoring.adoc:54]: Absolutely, and the implementation should do that.
To a degree, the issue is tamped by the job concurrency limit on the namespace. The pods run 5 at a time (even in the current implementation, which already does spawn them all at once)

**Pull request merged**